include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -227,15 +227,13 @@ namespace internal
     {
       for (unsigned int dataset = 0; dataset < dof_data.size(); ++dataset)
         {
-          bool duplicate = false;
-          for (unsigned int j = 0; j < dataset; ++j)
-            if (finite_elements[dataset].get() == finite_elements[j].get())
-              {
-                duplicate = true;
-                break;
-              }
-
-          if (duplicate == false)
+          const bool is_duplicate =
+            std::any_of(finite_elements.cbegin(),
+                        finite_elements.cbegin() + dataset,
+                        [&](decltype(*finite_elements.cbegin()) fe) {
```

Comment:
```suggestion
                        [&](const std::shared_ptr<dealii::hp::FECollection<dim, spacedim>> &fe) {
```

Reply:
Is there a specific reason for typing `dealii::hp::FECollection` instead of `hp::FECollection`?

Reply:
No reason -- just a copy-paste issue from when I looked up the declaration of the `finite_elements` variable. Go with the shorter version.

Reply:
Compiler is telling me that `dealii::` is mandatory because of name collision with `dealii::internal::hp::FECollection`.